core/src/main/java/io/smallrye/concurrency/impl/ManagedExecutorImpl.java

```diff
@@ -21,33 +21,37 @@
 	private ThreadContextImpl threadContext;
 
 	public ManagedExecutorImpl(int maxAsync, int maxQueued, ThreadContextImpl threadContext) {
-		super(0, maxAsync, 0l, null, new LinkedBlockingQueue<>(maxQueued));
+		super(maxAsync, maxAsync, 5000l, TimeUnit.MILLISECONDS,
+				new ArrayBlockingQueue<>(maxQueued, true), new ThreadPoolExecutor.AbortPolicy());
+		// we set core thread == max threads but allow for core thread timeout
+		// this prevents delaying spawning of new thread to when the queue is full
+		this.allowCoreThreadTimeOut(true);
```

Comment:
What are the reasons for this chance? I took those previous values from the JDK's default queues, so I thought they made sense.

Reply:
Right, so the problem here is that `ThreadPoolExecutor` by default creates one core thread and runs something on it. Then it puts _everything_ into queue and only spawns another thread when the queue is full. This is a problem especially with "unlimited" queue (and was obviously causing TCK failures).

What I did is to allow `coreThreads` == `maxThreads` which basically means eager initialization.
But to prevent threads idly hanging I also added `allowCoreThreadTimeOut()` and a timeout of 5s (not sure that is good time?) so that even core threads can be eliminated when not used for given period of time.

Reply:
Ah, I see. What's weird is that I had all tests running except shutdownNowPreventsAdditionalSubmitsAndCancelsTasks and maxAsync2. Did that fix those tests?

Reply:
`shutdown*` test is blocked by https://github.com/eclipse/microprofile-concurrency/issues/78
where TCK assumes different state of `Future`s which also requires custom implementation of futures and is IMO not correct.

`maxAsync2` is indeed fixed by this

Reply:
OK, I've put back your fix. Thanks for the explanation. The JDK docs are not clear about this at all, but in `Executors` I see usage of `LinkedBlockingQueue` is always with `core == max`, where `core = 0 < max` is using `SynchronousBlockingQueue` which has no max queue size. So your usage pattern is consistent with what the JDK is doing. Thanks.